shell/src/context_listener.rs

```diff
@@ -266,16 +214,17 @@ fn listen_protocol_events(
                     event_count + 1
                 };
 
-                if msg.perform {
-                    perform_context_action(&msg.action, context)?;
+                if msg.record {
+                    // record action in the order they are really comming
+                    for recorder in action_store_backend.iter_mut() {
+                        if let Err(error) = recorder.record(&msg) {
+                            error!(log, "action: {:?} ,error: {} ", &msg.action, error);
```

Comment:
@mat13mn 
please better message and error should fail something, warn we can continue
```
warn!(log, "Failed to store context action"; "action" => format!("{:?}", &msg.action), "reason" => format!("{}", error));
```